### ISSUE TITLE
Action list longer than 9 actions (single digits) does not work

### ISSUE THREAD

**Issue by bar0n36** [elastic/curator#1398]
When you provide an action list with greater than 9 actions (get to double digits), or you start the action numbering with a leading zero (01, 02 etc), curator fails to register the actions in the action file, until you get to an action where the first digit is 1.

### Expected Behavior
Curator should parse the action number, regardless of the number of digits.

### Actual Behavior
For a file list longer than 9, Curator will only register action 1/the last action (say we had 13 actions, the 13th action will be the only one parsed)

### Steps to Reproduce the Problem
1. Create an action file with a significant number of actions >9
2. Attempt to execute the action list
3. The only action that will execute is the last action with a leading 1
NOTE: I would expect that if there were 29 actions, Action 19 would execute, as would action 29, but I have not had chance to fully replicate this scenario

### Specifications

  - Version: 5.6
  - Platform: RHEL 6.9
  - Subsystem: Python 2.6.6

## Proposal
Curator should handle X number of Actions being listed in the action file.

**Comment by Leaf-Lin** [comment]
+1

**Comment by untergeek** [comment]
## Cannot replicate
I can't replicate what you're describing:

## Configuration
```
---
actions:
  1:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  2:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  3:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  4:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  5:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  6:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  7:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  8:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  9:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
  10:
    action: open
    filters:
    - filtertype: pattern
      kind: prefix
      value: index
```

## Output
```
2019-08-01 23:52:16,307 INFO      Preparing Action ID: 1, "open"
2019-08-01 23:52:16,307 INFO      Creating client object and testing connection
2019-08-01 23:52:16,309 INFO      Instantiating client object
2019-08-01 23:52:16,309 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,309 INFO      Testing client connectivity
2019-08-01 23:52:16,319 INFO      Trying Action ID: 1, "open": No description given
2019-08-01 23:52:16,332 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,335 INFO      Action ID: 1, "open" completed.
2019-08-01 23:52:16,335 INFO      Preparing Action ID: 2, "open"
2019-08-01 23:52:16,335 INFO      Creating client object and testing connection
2019-08-01 23:52:16,335 INFO      Instantiating client object
2019-08-01 23:52:16,335 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,335 INFO      Testing client connectivity
2019-08-01 23:52:16,341 INFO      Trying Action ID: 2, "open": No description given
2019-08-01 23:52:16,358 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,361 INFO      Action ID: 2, "open" completed.
2019-08-01 23:52:16,361 INFO      Preparing Action ID: 3, "open"
2019-08-01 23:52:16,361 INFO      Creating client object and testing connection
2019-08-01 23:52:16,361 INFO      Instantiating client object
2019-08-01 23:52:16,361 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,361 INFO      Testing client connectivity
2019-08-01 23:52:16,367 INFO      Trying Action ID: 3, "open": No description given
2019-08-01 23:52:16,383 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,388 INFO      Action ID: 3, "open" completed.
2019-08-01 23:52:16,388 INFO      Preparing Action ID: 4, "open"
2019-08-01 23:52:16,388 INFO      Creating client object and testing connection
2019-08-01 23:52:16,388 INFO      Instantiating client object
2019-08-01 23:52:16,388 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,389 INFO      Testing client connectivity
2019-08-01 23:52:16,396 INFO      Trying Action ID: 4, "open": No description given
2019-08-01 23:52:16,411 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,415 INFO      Action ID: 4, "open" completed.
2019-08-01 23:52:16,415 INFO      Preparing Action ID: 5, "open"
2019-08-01 23:52:16,415 INFO      Creating client object and testing connection
2019-08-01 23:52:16,415 INFO      Instantiating client object
2019-08-01 23:52:16,415 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,415 INFO      Testing client connectivity
2019-08-01 23:52:16,423 INFO      Trying Action ID: 5, "open": No description given
2019-08-01 23:52:16,440 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,443 INFO      Action ID: 5, "open" completed.
2019-08-01 23:52:16,443 INFO      Preparing Action ID: 6, "open"
2019-08-01 23:52:16,443 INFO      Creating client object and testing connection
2019-08-01 23:52:16,443 INFO      Instantiating client object
2019-08-01 23:52:16,443 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,443 INFO      Testing client connectivity
2019-08-01 23:52:16,449 INFO      Trying Action ID: 6, "open": No description given
2019-08-01 23:52:16,470 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,473 INFO      Action ID: 6, "open" completed.
2019-08-01 23:52:16,473 INFO      Preparing Action ID: 7, "open"
2019-08-01 23:52:16,473 INFO      Creating client object and testing connection
2019-08-01 23:52:16,473 INFO      Instantiating client object
2019-08-01 23:52:16,474 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,474 INFO      Testing client connectivity
2019-08-01 23:52:16,479 INFO      Trying Action ID: 7, "open": No description given
2019-08-01 23:52:16,498 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,501 INFO      Action ID: 7, "open" completed.
2019-08-01 23:52:16,501 INFO      Preparing Action ID: 8, "open"
2019-08-01 23:52:16,502 INFO      Creating client object and testing connection
2019-08-01 23:52:16,502 INFO      Instantiating client object
2019-08-01 23:52:16,502 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,502 INFO      Testing client connectivity
2019-08-01 23:52:16,507 INFO      Trying Action ID: 8, "open": No description given
2019-08-01 23:52:16,524 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,527 INFO      Action ID: 8, "open" completed.
2019-08-01 23:52:16,527 INFO      Preparing Action ID: 9, "open"
2019-08-01 23:52:16,527 INFO      Creating client object and testing connection
2019-08-01 23:52:16,528 INFO      Instantiating client object
2019-08-01 23:52:16,528 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,528 INFO      Testing client connectivity
2019-08-01 23:52:16,534 INFO      Trying Action ID: 9, "open": No description given
2019-08-01 23:52:16,547 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,551 INFO      Action ID: 9, "open" completed.
2019-08-01 23:52:16,551 INFO      Preparing Action ID: 10, "open"
2019-08-01 23:52:16,551 INFO      Creating client object and testing connection
2019-08-01 23:52:16,551 INFO      Instantiating client object
2019-08-01 23:52:16,552 INFO      Created Elasticsearch client object with provided settings
2019-08-01 23:52:16,552 INFO      Testing client connectivity
2019-08-01 23:52:16,559 INFO      Trying Action ID: 10, "open": No description given
2019-08-01 23:52:16,574 INFO      Opening 1 selected indices: ['index']
2019-08-01 23:52:16,577 INFO      Action ID: 10, "open" completed.
2019-08-01 23:52:16,577 INFO      Job completed.
```

## Conclusion

I believe you are experiencing this problem because of a _very_ outdated version of Python: `2.6.6`, according to what you shared. Curator is only expected to work in 2.7.9 and up. The above result tested identically with Python 2.7.15 and 3.6.8.

A proposed solution is to use the available [YUM repository](https://www.elastic.co/guide/en/elasticsearch/client/curator/5.7/yum-repository.html) or even directly download the [Curator 5.7.6 RPM](https://packages.elastic.co/curator/5/centos/6/Packages/elasticsearch-curator-5.7.6-1.x86_64.rpm) for CentOS/RedHat 6.

Another is to install with Python 3.5 or higher.

**Comment by bar0n36** [comment]
That was the theory I had just come to as well, given you have failed to replicate both issues I have raised. I can confirm Python 2.6.6 is in use, which I agree is vastly outdated, but is what I have on the platforms in question. I will look at my options to upgrade.

**Comment by untergeek** [comment]
@bar0n36 The aforementioned RPMs come bundled with their own libs, which are frozen Python 3.6. It won't interfere with the system python this way, and it should just work.

**Comment by bar0n36** [comment]
@untergeek . Then in that case, that theory does not apply.
elasticsearch-curator-5.6.0-1.x86_64 is installed. Happy to upgrade to 5.7 to verify, but it has always been installed from RPM.

**Comment by bar0n36** [comment]
Leave it with me. I will upgrade Curator to latest 5.7 from RPM and verify both issues I have reported are addressed in the 5.7 branch.
